.pre-commit-config.yaml

```diff
@@ -0,0 +1,52 @@
+# If you want use pre-commit, you need to install pre-commit package.
+# $ pip install pre-commit
+# $ pre-commit install
+
+default_language_version:
+  python: python3
+
+repos:
+  - repo: https://github.com/psf/black
+    rev: 24.4.0
+    hooks:
+      - id: black
+  - repo: https://github.com/pycqa/flake8
+    rev: 7.0.0
+    hooks:
+      - id: flake8
+        exclude: .venv|venv|build|tutorial|.asv
+        args: [
+            "--max-line-length=99",
+            "--ignore=E203,E704,W503",
+            "--statistics",
+        ]
```

Comment:
For flake8, we need to explicitly use [pyproject-flake8](https://github.com/microsoft/vscode-flake8/issues/135) to migrate this to pyproject.toml, but IMO, it is worth discussing if we should replace flake8 with pyproject-flake8.